prov/efa/src/efa_prov_info.c

```diff
@@ -158,40 +164,35 @@ const struct fi_ep_attr efa_ep_attr = {
  *
  * @param	prov_info[out]		prov_info pointer
  * @param	device[in]		pointer to an efa_device struct, which contains device attributes
- * @param	ep_type			endpoint_type, can be FI_EP_RDM or FI_EP_DGRAM
+ * @param	info_type			EFA info type
  */
 static
 void efa_prov_info_set_ep_attr(struct fi_info *prov_info,
 			       struct efa_device *device,
-			       enum fi_ep_type ep_type)
+			       enum efa_info_type info_type)
 {
 
 	*prov_info->ep_attr = efa_ep_attr;
-	if (ep_type == FI_EP_DGRAM) {
-		prov_info->mode |= FI_MSG_PREFIX;
-		prov_info->ep_attr->msg_prefix_size = 40;
-	}
-
 	prov_info->ep_attr->protocol	= FI_PROTO_EFA;
-	prov_info->ep_attr->type	= ep_type;
-
-	if (prov_info->ep_attr->type == FI_EP_RDM) {
-		prov_info->tx_attr->inject_size = device->efa_attr.inline_buf_size;
-	} else {
-		assert(prov_info->ep_attr->type == FI_EP_DGRAM);
-                /*
-		 * Currently, there is no mechanism for device to discard
-		 * a completion, therefore there is no way for dgram endpoint
-		 * to implement FI_INJECT. Because FI_INJECT is not an optional
-		 * feature, we had to set inject_size to 0.
-		 * 
-		 * TODO:
-		 * Remove this after implementing cq read for efa-raw
-                 */
-		prov_info->tx_attr->inject_size = 0;
+	prov_info->ep_attr->max_msg_size = device->ibv_port_attr.max_msg_sz;
+
+	switch (info_type) {
+		case EFA_INFO_RDM:
+			prov_info->ep_attr->type = FI_EP_RDM;
+			/* max_msg_size for RDM path is set in efa_prov_info_alloc_for_rdm */
+			break;
+		case EFA_INFO_DIRECT:
+			prov_info->ep_attr->type = FI_EP_RDM;
+			/* max_msg_size in ep_attr should be the maximum of send/recv and RMA sizes */
+			prov_info->ep_attr->max_msg_size = MAX(device->ibv_port_attr.max_msg_sz, device->max_rdma_size);
```

Comment:
I missed one thing earlier. We should only do the `MAX(device->ibv_port_attr.max_msg_sz, device->max_rdma_size)` when hints request FI_RMA and the efa-direct can supports it, which means the device needs to support read, write, and unsolicited write recv

Reply:
And you need to add `FI_RMA` cap into efa-direct's prov_info when device supports it. Today if I run with efa-direct for your PR I will get a failed fi_getinfo for `fi_rma_bw` which requests `FI_RMA`

```
ce/libfabric/fabtests$ FI_LOG_LEVEL=warn fi_rma_bw -p efa -f efa-direct -E -S 1024
fi_getinfo(): common/shared.c:1071, ret=-61 (No data available)
(env) ubuntu@ip-172-31-40-58:~/PortaFiducia/build/libraries/libfabric/main/source/libfabric/fabtests$ 
```